lab-jaren/test/game-router-test.js

```diff
@@ -0,0 +1,150 @@
+    });
+    it('valid input should create a game with status 200', done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send(mockData)
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.title).to.equal('XCOM 2');
+        expect(res.body.genre).to.equal('strategy/tactics');
+        expect(res.body.developer).to.equal('2K Games, Inc.');
+        expect(res.body.publisher).to.equal('Firaxis Games, Inc.');
+        expect(res.body.platforms).to.equal('Windows, Linux, Mac, PS4, Xbox One');
+        expect(res.body.ratingESRB).to.equal('Teen');
+        expect(res.body.releaseDate).to.equal('Feb 05, 2016');
+        expect(Boolean(res.body.id)).to.equal(true);
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    it('invalid input should return 400 status code', done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send({title: 'I am bread'})
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(400);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing GET /api/games/:id', function() {
+    before(done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send(mockData)
+      .then(res => {
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    after(done => {
+      Game.deleteById(this.tempGameId)
+      .then(() => done())
+      .catch(done);
+    });
+    it('valid id should return a game with status 200', done => {
+      superagent.get(`${baseURL}/api/games/${this.tempGameId}`)
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.title).to.equal('XCOM 2');
+        expect(res.body.genre).to.equal('strategy/tactics');
+        expect(res.body.developer).to.equal('2K Games, Inc.');
+        expect(res.body.publisher).to.equal('Firaxis Games, Inc.');
+        expect(res.body.platforms).to.equal('Windows, Linux, Mac, PS4, Xbox One');
+        expect(res.body.ratingESRB).to.equal('Teen');
+        expect(res.body.releaseDate).to.equal('Feb 05, 2016');
+        expect(Boolean(res.body.id)).to.equal(true);
+        done();
+      })
+      .catch(done);
+    });
+    it('invalid id should return status 404', done => {
+      superagent.get(`${baseURL}/api/games/42`)
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(404);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing GET /api/games', function() {
+    before(done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send(mockData)
+      .then(res => {
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    after(done => {
+      Game.deleteById(this.tempGameId)
+      .then(() => done())
+      .catch(done);
+    });
+    it('should return an array containing .gitignore plus the IDs and status 200', done => {
```

Comment:
Good job with the bonus. I wrote a way you can get rid of ``.gitignore`` from the array if you want to utilize it (because from an API requester's standpoint, that's a little weird to see that filename returned to them). 

```
storage.availableIDs = function(schema) {
  debug('availableIDs()');
  return fs.readdirProm(`${dataDir}/${schema}`)
  .catch(err => Promise.reject(createError(404, err.message)))
  .then(filenames => {
    // filenames = filenames.map(name => name.split('.json')[0]);
    filenames = filenames.map((name) => {
      name = name.split('.json')[0];
      return name;
    }).filter((element) => {
      return element !== '.gitignore';
    });
    return filenames;
  });
};
```